packages/assets-controllers/src/TokensController.ts

```diff
@@ -960,6 +961,24 @@ export class TokensController extends BaseController<
       );
     }
   }
+
+  addPendingApprovals(updateBadge: () => void) {
```

Comment:
It will be better to return a promise instead of callback like I originally did. 
Ref thread: https://github.com/MetaMask/metamask-extension/pull/18796#discussion_r1182350649